File: 13-04-2026/palindrome_char.py
def palindrome_locator(s):
    strToArr = list(s)
    oddOrEven = "".join(strToArr[int(len(s) / 2 - 1) : int(len(s) / 2 + 1)]) if s and len(s) % 2 == 0 else s[len(s) // 2]

    return oddOrEven if s[::-1] == s else "none";


print(palindrome_locator("racecar")); # "e".
print(palindrome_locator("level")); # "v".
print(palindrome_locator("freecodecamp")); # "none".
print(palindrome_locator("noon")); # "oo".
print(palindrome_locator("11100111")); # "00".

File: 13-04-2026/palindrome_char.js
function palindromeLocator(str) {
  const oddOrEven =
    str && str.length % 2 === 0
      ? str.slice(str.length / 2 - 1, str.length / 2 + 1)
      : str[Math.floor(str.length / 2)];

  return str.split("").reverse().join("") === str ? oddOrEven : "none";
}

console.log(palindromeLocator("racecar")); // "e".
console.log(palindromeLocator("level")); // "v".
console.log(palindromeLocator("freecodecamp")); // "none".
console.log(palindromeLocator("noon")); // "oo".
console.log(palindromeLocator("11100111")); // "00".
